src/main/scala/strawman/collection/IterableFactory.scala

```diff
@@ -0,0 +1,35 @@
+package strawman
+package collection
+
+import strawman.collection.mutable.Builder
+
+import scala.Int
+
+/** Base trait for instances that can construct a collection from an iterable */
+trait FromIterable[+C[X] <: Iterable[X]] {
+  def fromIterable[B](it: Iterable[B]): C[B]
+}
+
+/** Base trait for builder factories */
+trait CanBuild[-A, +Repr] {
```

Comment:
Alright, I was reluctant to define implicit vals with type `Function1` because they can act as implicit conversions, but in the case the type `() => _` the corresponding type is `Function0` so it’s perfectly fine.